### PR TITLE
fix hcat ambiguity errors on julia master (v0.6) for *nix

### PR DESCRIPTION
**Pull request by cjprybol**
https://github.com/JuliaData/DataTables.jl/pull/18#issuecomment-281494609

### CODE REVIEW

File: src/datatable/datatable.jl [hunk 1]
```diff
@@ -764,8 +764,16 @@ hcat!(dt::DataTable, x) = hcat!(dt, DataTable(Any[NullableArray([x])]))
...
 hcat!(a::DataTable, b, c...) = hcat!(hcat!(a, b), c...)
 
-# hcat
+# hcat for 2 arguments
 Base.hcat(dt::DataTable, x) = hcat!(copy(dt), x)
+Base.hcat(dt1::DataTable, dt2::AbstractDataTable) = hcat!(copy(dt1), dt2)
+
+# hcat! for >2 arguments
+function Base.hcat(a::DataTable,
+          b::AbstractDataTable,
```
Comment (nalimilan) [inline src/datatable/datatable.jl]: Please align arguments.

**Comment by nalimilan** [comment]
Acutally, I think a simpler fix would be to just define `hcat(::DataTable, ::DataTable)`, which must be the problematic case.

**Comment by nalimilan** [comment]
See corresponding change at https://github.com/JuliaStats/DataFrames.jl/pull/1164/commits/64615ddd1c7720c490aacb3355bd78a344e7c21a#diff-6108c932ccdff77aea536b99a80d4a79 and https://github.com/JuliaStats/DataFrames.jl/pull/1164/commits/64615ddd1c7720c490aacb3355bd78a344e7c21a#diff-0aa1ff52d56badabf7d1511d3dc3f279

**Comment by cjprybol** [comment]
I needed to make the 2 argument and 3-or-more argument cases in order to squash the ambiguity error but I can see that those are both taken care of [here](https://github.com/JuliaStats/DataFrames.jl/pull/1164/commits/64615ddd1c7720c490aacb3355bd78a344e7c21a#diff-6108c932ccdff77aea536b99a80d4a79R683) and [here](https://github.com/JuliaStats/DataFrames.jl/pull/1164/commits/64615ddd1c7720c490aacb3355bd78a344e7c21a#diff-6108c932ccdff77aea536b99a80d4a79R686)! Thanks for the links. Should I close this pull request or modify it to match the DataFrames pull? In general, what's the protocol for how we want to maintain the two packages in parallel? All essential fixes are made in DataFrames and then `git cherry-pick`ed into DataTables? Anything that isn't essential for keeping DataFrames alive through Julia 0.6 is made here in DataTables?

File: src/datatable/datatable.jl [hunk 2]
```diff
@@ -764,8 +764,16 @@ hcat!(dt::DataTable, x) = hcat!(dt, DataTable(Any[NullableArray([x])]))
 hcat!(dt::DataTable) = dt
 hcat!(a::DataTable, b, c...) = hcat!(hcat!(a, b), c...)
 
-# hcat
+# hcat for 2 arguments
 Base.hcat(dt::DataTable, x) = hcat!(copy(dt), x)
+Base.hcat(dt1::DataTable, dt2::AbstractDataTable) = hcat!(copy(dt1), dt2)
+
+# hcat! for >2 arguments
+function Base.hcat(dt1::AbstractDataTable,
```
Comment (nalimilan) [inline src/datatable/datatable.jl]: This method shouldn't be defined in this file as it applies to `AbstractDataTable`.

File: src/datatable/datatable.jl [hunk 3]
```diff
@@ -764,8 +764,16 @@ hcat!(dt::DataTable, x) = hcat!(dt, DataTable(Any[NullableArray([x])]))
 hcat!(dt::DataTable) = dt
 hcat!(a::DataTable, b, c...) = hcat!(hcat!(a, b), c...)
 
-# hcat
+# hcat for 2 arguments
 Base.hcat(dt::DataTable, x) = hcat!(copy(dt), x)
+Base.hcat(dt1::DataTable, dt2::AbstractDataTable) = hcat!(copy(dt1), dt2)
+
+# hcat! for >2 arguments
```
Comment (nalimilan) [inline src/datatable/datatable.jl]: I don't think this comment is really needed, it's quite clear from the signature already.

**Comment by nalimilan** [comment]
> I needed to make the 2 argument and 3-or-more argument cases in order to squash the ambiguity error but I can see that those are both taken care of here and here! Thanks for the links. Should I close this pull request or modify it to match the DataFrames pull?

Updating a PR is fine.

> In general, what's the protocol for how we want to maintain the two packages in parallel? All essential fixes are made in DataFrames and then git cherry-picked into DataTables? Anything that isn't essential for keeping DataFrames alive through Julia 0.6 is made here in DataTables?

I don't think it's completely clear yet. At least for small fixes, it's better to apply them to both projects. Larger change sets like the grouping PR are quite hard to adapt, so maybe we can keep it for DataTables.

**Comment by nalimilan** [comment]
Thanks!

**Pull request merged**